File: src/ScotlandsMountains.Import.ConsoleApp/Entities/Classification.cs
namespace ScotlandsMountains.Import.ConsoleApp.Entities;

public class Classification : Entity
{
    public int? DisplayOrder { get; set; }

    public bool IsActive { get; set; }
    
    public string NameSingular { get; set; } = null!;
    
    public string Description { get; set; } = null!;

    public List<MountainClassification> Mountains { get; set; } = new();
}

File: src/scotlands_mountains_app/lib/models/classification.dart
class Classification {
  final int id;
  final int? displayOrder;
  final int isActive;
  final String nameSingular;
  final String description;
  final String name;

  const Classification._(
    this.id,
    this.name,
    this.displayOrder,
    this.isActive,
    this.nameSingular,
    this.description,
  );

  factory Classification(Map<String, dynamic> map) {
    return Classification._(map['id'], map['name'], map['displayOrder'],
        map['isActive'], map['nameSingular'], map['description']);
  }
}
